tests/registry.test.ts

```diff
@@ -0,0 +1,333 @@
+import { assert, expect } from "@esm-bundle/chai";
+import { init, initializeLogger, Wasmer } from "..";
+
+const initialized = (async () => {
+	await init(new URL("../dist/wasmer_js_bg.wasm", import.meta.url), undefined,
+		{ registry_url: "https://registry.wasmer.wtf/graphql", token: "<YOUR_TOKEN>" }
+	);
+	initializeLogger("error");
+})();
+
+
+const owner = "YOUR_NAME"
+const pkg_name = "test-js-sdk"
+const app_name = "test-js-sdk"
+
+describe("Registry", function() {
+	this.timeout("60s").beforeAll(async () => await initialized);
+
+	it("Has global context", async () => {
+		let v = (globalThis as any)["__WASMER_REGISTRY__"];
+		expect(typeof v != "undefined")
+	});
+
+
+	it("can create a package with atoms", async () => {
+		let manifest =
+		{
+			"module": [
+				{
+					"name": "test",
+					"abi": "wasi",
+					"source": new Uint8Array([0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00])
+				},
+				{
+					"name": "other-test",
+					"source": new Uint8Array([0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00])
+				}
+			],
+			"command": [
+				{
+					"module": "wasmer/python:python",
+					"name": "hello",
+					"runner": "wasi",
+					"annotations": {
+						"wasi": {
+							"main-args": [
+								"-c",
+								"import os; print([f for f in os.walk('/public')]); "
+							]
+						}
+					}
+				}
+			],
+			"dependencies": {
+				"wasmer/python": "3.12.5+build.7"
+			},
+
+			"fs": {
+				"public": {
+					"index.html": "Hello, js!",
+					"index_timestamp.html": { data: "Hello, js!", modified: 987656789 },
+					"index_date.html": { data: "Hello, js!", modified: new Date() }
+				}
+			}
+		}
+		let result = await Wasmer.createPackage(manifest);
+		console.log("\n== Creating a package == ");
+		console.log(result.commands)
+	})
+
+
+	it("can create a package", async () => {
+		let manifest =
+		{
+			"command": [
+				{
+					"module": "wasmer/python:python",
+					"name": "hello",
+					"runner": "wasi",
+					"annotations": {
+						"wasi": {
+							"main-args": [
+								"-c",
+								"import os; print([f for f in os.walk('/public')]); "
+							]
+						}
+					}
+				}
+			],
+			"dependencies": {
+				"wasmer/python": "3.12.5+build.7"
+			},
+
+			"fs": {
+				"public": {
+					"index.html": "Hello, js!",
+					"index_timestamp.html": { data: "Hello, js!", modified: 987656789 },
+					"index_date.html": { data: "Hello, js!", modified: new Date() }
+				}
+			}
+		}
+		let result = await Wasmer.createPackage(manifest);
+		console.log("\n== Creating a package == ");
+		console.log(result.commands)
+	})
+
+
+
+	it("can publish packages", async () => {
+		let manifest =
+		{
+			"package": {
+				"name": owner + "/" + pkg_name
```

Comment:
Does it need to be named? Do we support unnamed ones? If so, are we testing as well unnamed packages?

Reply:
The test above, as you prompted, creates an unnamed package.

Reply:
Ah, I got confused between creating and publishing. No, publishing unnamed packages was not supported. We can support it by fetching the namespace from the given token and use it as a namespace. Otherwise, users will have to provide it either in the call to `publishPackage` or in the manifest description itself.

Reply:
It's actually possible to have unnamed packages per se, users can specify a package manifest structured like this: 
```js
let manifest = {
  "package": { "name": "edoardo/" },
   "command": [
	{
	 "module": "php/php:php",
	"name": "run",
    ...
```

Reply:
Umm... we should support unnamed packages fully (without requiring any name), as we do in the CLI @xdoardo

Reply:
done as prompted